src/error/mod.rs

```diff
@@ -25,5 +30,42 @@ pub fn handler(e: CalcError) -> String {
         },
         CalcError::Syntax(details) => format!("Syntax Error: {}", details),
         CalcError::Parser(details) => format!("Parser Error: {}", details),
+        CalcError::Help => format!(
+            "Constants\n{}\nFunctions\n{}\nOperators\n{}\n",
+            blocks(lex::CONSTANTS.keys().cloned()),
+            blocks(lex::FUNCTIONS.keys().cloned()),
+            {
+                let l: Vec<_> = lex::OPERATORS.keys().map(|c| c.to_string()).collect();
+                l.join(" ")
+            }
+        ),
+    }
+}
+
+/// Convert iterator into strings of chunks of 8 right padded with space.
+fn blocks(mut iter: impl Iterator<Item = &'static str> + ExactSizeIterator) -> String {
+    // calculate max width but ideally this should be calculated once
+    let mut max_width = 79; // capped at 79
+    if let Ok(s) = std::env::var("COLUMNS") {
```

Comment:
Environment variable seemed lost here, not sure why.

Reply:
`$COLUMNS` and `$ROWS` are shell variables and are usually available only to bash, other programs can't access them unless they are manually exported (I believe that is the reason). We could perhaps use another crate to fetch terminal size, or if that feels too contrived, simply run a `std::process::Command`: `stty size`.

Reply:
I got some error running `stty size`.
```
[src/error/mod.rs:41] output = Output {
    status: ExitStatus(
        ExitStatus(
            256,
        ),
    ),
    stdout: "",
    stderr: "stty: \'standard input\': Inappropriate ioctl for device\n",
}
```
```rust
            // calculate max width but ideally this should be calculated once
            let output = Command::new("stty")
                .arg("size")
                .output()
                .expect("failed to get `stty size`");
            let mut max_width = 79; // capped at 79
            if let Some(b) = dbg!(output).stdout.splitn(2, |&c| c == b' ').nth(1) {
                if let Ok(s) = std::str::from_utf8(b) {
                    if let Ok(n) = s.parse() {
                        if n < max_width {
                            max_width = n;
                        }
                    }
                }
            }
```

Reply:
I guess I will just use `term_size` crate for this, it also works on windows but this method does not.